초기 Entity 및 값타입 작성

ffinn92 opened the pull request:
### ✔️이슈번호
  - Issue #6

### 📚작업목록
  - [x] Entity 및 값타입 작성
  - [x] Application 이름 수정
  - [x] BaseTime 클래스 작성 
 

### 📖작업내용
  -  Member, Crew, CrewApplication Entity와 그에 따른 값타입들을 작성해봤습니다.

### 🙋‍♂️참고사항
  - ERD 초안에는 CrewApplication이 없었지만 요청사항들을 모아두기 위해 필요할것 같아 추가하였습니다.
  - BaseTime은 생성, 수정 기능이 다른 domain에서도 필요하므로 Common 패키지 항목에 추가하였습니다.
  - 이전에 Name.java 값타입 파일이 Git에 add 되지않아 테스트를 통과하지 못하여 다시 PR 날립니다.

ffinn92 committed:
Chore: ERD 추출을 위한 1차 Entity 작성

ffinn92 committed:
Refactor: Application.java Entity 이름 변경

- Application.java -> CrewApplication.java 변경

ffinn92 committed:
Chore: Name.java 값타입 Git add 진행

ffinn92 committed:
Chore: Index.adoc 파일 추가 commit

NB993 commented on src/main/java/com/calefit/crew/domain/Description.java:
```diff
@@ -0,0 +1,14 @@
+
+@Getter
+@NoArgsConstructor
+@Embeddable
```
Crew나 Member엔티티에 멤버변수가 하나만 있는 값타입 클래스가 자주 사용되고 있는데 혹시 일반 멤버변수로 선언하지 않고 값타입으로 선언하신 이유가 있으신가요??

ffinn92 replied:
설명글이나, 닉네임등 입력할때 길이나 valid 처리할때 값타입 내에서 처리하기 위함입니다.

NB993 replied:
유효성 검사만을 위해서 따로 분리하는 대신 아래 사진과 같이 엔티티 멤버변수에 직접 어노테이션으로 유효성 검사를 하는 건 어떻게 생각하시나요? 저는 임베디드 타입이 둘 이상의 변수에 공통적인 성격이 존재할 때 이를 묶어내는 용도로 사용하는 것이 조금 더 적절하다고 생각했습니다..!
https://chiki-cha.tistory.com/m/150
![image](https://user-images.githubusercontent.com/46119135/184328376-ebda0dbf-9481-46dc-b958-f7f7594070eb.png)

ffinn92 replied:
네 알겠습니다. 임베디드는 말씀 주신대로 둘 이상의 공통적인 성격이 존재할때 쓰도록 하겠습니다.
다만, 현재 있는 유효성검사 어노테이션들로는 저희가 원하는 유효성 검사를 100% 하기는 어려울것 같기도 하고, 나중에 추가적인 유효성검사 로직이 필요할 수도 있기 때문에 Custom된 유효성검사 어노테이션을 사용하는것은 어떠신가요?

https://dotoridev.tistory.com/4

NB993 replied:
넵 좋은 방법인 것 같습니다 :)

NB993 commented on src/main/java/com/calefit/crew/entity/CrewApplication.java:
```diff
@@ -0,0 +1,22 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
```
여기도 LAZY 걸어주면 좋을 것 같습니닷

ffinn92 replied:
넵! 확인 감사합니다.

NB993 commented on src/main/java/com/calefit/member/entity/Member.java:
```diff
@@ -0,0 +1,44 @@
+    private String role;
+
+    @ManyToOne(fetch = LAZY)
+    @JoinColumn(name = "crew_id")
```
혹시 ManyToOne에서 JoinColumn을 생략하는 방법은 어떠신가요?
https://hyeon9mak.github.io/omit-join-column-when-using-many-to-one/

ffinn92 replied:
네 좋습니다 좋은 자료 공유 감사합니다. 생략하는게 더 깔끔할것 같네요.

NB993 commented on src/main/java/com/calefit/crew/entity/Crew.java:
```diff
@@ -0,0 +1,39 @@
+    @OneToMany(mappedBy = "member")
+    private List<Member> members = new ArrayList<>();
+
+    private Long member_cnt;
```
스네이크 케이스로 선언되어있네요!

ffinn92 replied:
네 다음 PR때 수정하도록 하겠습니다. 피오

NB993 reviewed (commented):
고생하셨습니다 반스! 저도 얼른 작성해서 pr올리도록 하겠습니다 :)

ffinn92 committed:
Refactor: PR 리뷰 반영

- Crew 클래스 내 스네이크 케이스 선언 수정 및 mappedBy = "crew"로 수정
- CrewApplication LAZY 추가
- Member 클래스 내 JoinColumn 생략

ffinn92 committed:
Refactor: CrewApplication 클래스 내 JoinColumn 생략
